### PR TITLE
Make version field a text box to allow for semantic version entries

### PR DESCRIPTION
**Pull request by leefaisonr**
ref #1935 

Text box allows for semantic versioning:
Screenshot 2024-09-24 at 9 37 03 AM

### CODE REVIEW

File: spec/system/curator_controlled_metadata_spec.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@
       expect(page).to have_field("ark", readonly: true)
       expect(page).to have_field("resource_type", readonly: true)
       expect(page).to have_field("resource_type_general", disabled: true)
-      expect(page).to have_field("version_number", disabled: true)
+      expect(page).to have_field("version_number", disabled: false)
```
Comment (carolyncole) [inline spec/system/curator_controlled_metadata_spec.rb:27]: @leefaisonr I don't think the Princeton Submitter should be able to change the version number.

File: app/views/works/_curator_controlled_form.html.erb [hunk 2]
```diff
@@ -48,15 +48,12 @@
...
 <div class="field">
-  Version: <span class="required-field required-field-star">*</span><br>
-  <input type="hidden" name="version_number" value="<%= @work.resource.version_number%>"/> <!-- put the value in the form as disabled will not send the data back -->
-  <select id="version_number" name="version_number" class="input-version" <%= is_group_admin ? '' : 'disabled' %> >
-    <% (1..10).each do |i| %>
-      <option value="<%= i %>" <%= i.to_s == @work.resource.version_number ? "selected" : "" %>><%= i %></option>
-    <% end %>
-  </select>
+  Version: <br>
+  <input type="text" id="version_number" name="version_number" class="input-text-long" value="<%= @work.resource.version_number %>" <%= is_group_admin ? '' : 'readonly' %> />
```
Comment (carolyncole) [inline app/views/works/_curator_controlled_form.html.erb:52]: I think you still need something like `<%= is_group_admin ? '' : 'disabled' %> ` here

**Review by carolyncole — CHANGES_REQUESTED** [review]
Nice work @leefaisonr.  One small change is needed

File: spec/system/curator_controlled_metadata_spec.rb (hunk 1 above)
Comment (leefaisonr) [reply spec/system/curator_controlled_metadata_spec.rb]: Thanks @carolyncole - I have made those changes.

**Review by carolyncole — DISMISSED** [review]

File: app/views/works/_curator_controlled_form.html.erb [hunk 3]
```diff
@@ -48,15 +48,12 @@
...
 <!-- Version number -->
 <div class="field">
-  Version: <span class="required-field required-field-star">*</span><br>
-  <input type="hidden" name="version_number" value="<%= @work.resource.version_number%>"/> <!-- put the value in the form as disabled will not send the data back -->
-  <select id="version_number" name="version_number" class="input-version" <%= is_group_admin ? '' : 'disabled' %> >
-    <% (1..10).each do |i| %>
-      <option value="<%= i %>" <%= i.to_s == @work.resource.version_number ? "selected" : "" %>><%= i %></option>
-    <% end %>
-  </select>
+  Version: <br>
```
Comment (carolyncole) [inline app/views/works/_curator_controlled_form.html.erb:51]: @leefaisonr did you intentionally removed the required `<span class="required-field required-field-star">*</span>`?  I approved the PR incase you did.  Sorry I missed this earlier.

**Review by carolyncole — APPROVED** [review]
Thanks @leefaisonr

**Pull request merged**